packages/relay/environment.ts

```diff
@@ -19,12 +22,24 @@ export async function networkFetch(
   request: RequestParameters,
   variables: Variables
 ): Promise<GraphQLResponse> {
+  const headers: Dictionary<string> = {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+  };
+
+  // pasaport wraps next-auth and uses its cookie auth mechanism to validate sessions.
+  // on browsers, fetch request automatically attaches the existing cookies
+  // but on next.js servers, `fetch` is pollyfilled, and does not pick up cookies
+  // we use `next/header`'s cookies() helper here
+  if (IS_SERVER) {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const nextHeaders = require("next/headers") as { cookies: () => ReadonlyRequestCookies };
+    headers.Cookie = nextHeaders.cookies().toString();
+  }
```

Comment:
cc @cansirin this should fix your problem for getting "null" sessions for server side requests.